R/RunAnalyses.R

```diff
@@ -892,100 +920,118 @@ applyTrimMatchStratify <- function(ps, arguments) {
+  })
   return(NULL)
 }
 
 doPrefilterCovariates <- function(params) {
-  cohortMethodData <- loadCohortMethodData(params$cohortMethodDataFile)
-  covariates <- cohortMethodData$covariates
-  if (nrow_temp(covariates) > 0) {
-    if (params$args$useCovariates) {
-      covariatesToInclude <- params$args$includeCovariateIds
-      covariatesToExclude <- params$args$excludeCovariateIds
-    } else {
-      covariatesToInclude <- c()
-      covariatesToExclude <- c()
-    }
-    covariatesToInclude <- unique(c(covariatesToInclude, params$args$interactionCovariateIds))
-    if (length(covariatesToInclude) != 0) {
-      covariates <- covariates %>%
-        filter(.data$covariateId %in% covariatesToInclude)
-    }
-    if (length(covariatesToExclude) != 0) {
-      covariates <- covariates %>%
-        filter(!.data$covariateId %in% covariatesToExclude)
+  tryCatch({
+    cohortMethodData <- loadCohortMethodData(params$cohortMethodDataFile)
+    covariates <- cohortMethodData$covariates
+    if (nrow_temp(covariates) > 0) {
+      if (params$args$useCovariates) {
+        covariatesToInclude <- params$args$includeCovariateIds
+        covariatesToExclude <- params$args$excludeCovariateIds
+      } else {
+        covariatesToInclude <- c()
+        covariatesToExclude <- c()
+      }
+      covariatesToInclude <- unique(c(covariatesToInclude, params$args$interactionCovariateIds))
+      if (length(covariatesToInclude) != 0) {
+        covariates <- covariates %>%
+          filter(.data$covariateId %in% covariatesToInclude)
+      }
+      if (length(covariatesToExclude) != 0) {
+        covariates <- covariates %>%
+          filter(!.data$covariateId %in% covariatesToExclude)
+      }
     }
-  }
-  filteredCohortMethodData <- Andromeda::andromeda(
-    cohorts = cohortMethodData$cohorts,
-    outcomes = cohortMethodData$outcomes,
-    covariates = covariates,
-    covariateRef = cohortMethodData$covariateRef,
-    analysisRef = cohortMethodData$analysisRef
-  )
-  attr(filteredCohortMethodData, "metaData") <- attr(cohortMethodData, "metaData")
-  class(filteredCohortMethodData) <- "CohortMethodData"
-  attr(class(filteredCohortMethodData), "package") <- "CohortMethod"
-  saveCohortMethodData(filteredCohortMethodData, params$prefilteredCovariatesFile)
+    filteredCohortMethodData <- Andromeda::andromeda(
+      cohorts = cohortMethodData$cohorts,
+      outcomes = cohortMethodData$outcomes,
+      covariates = covariates,
+      covariateRef = cohortMethodData$covariateRef,
+      analysisRef = cohortMethodData$analysisRef
+    )
+    attr(filteredCohortMethodData, "metaData") <- attr(cohortMethodData, "metaData")
+    class(filteredCohortMethodData) <- "CohortMethodData"
+    attr(class(filteredCohortMethodData), "package") <- "CohortMethod"
+    saveCohortMethodData(filteredCohortMethodData, params$prefilteredCovariatesFile)
+  }, error = function(err) {
+    ParallelLogger::logError(err)
```

Comment:
This code has no effect. Errors are automatically caught by ParalllelLogger. Just rethrowing the error will achieve the same thing